fakeredis/_server.py

```diff
@@ -889,6 +889,16 @@ def _scan(self, keys, cursor, *args):
                 count = Int.decode(args[i + 1])
                 if count <= 0:
                     raise SimpleError(SYNTAX_ERROR_MSG)
+            elif casematch(args[i], b'type'):
+                type = args[i + 1]
+                is_valid = (casematch(type, b'string') or
+                            casematch(type, b'list') or
+                            casematch(type, b'set') or
+                            casematch(type, b'zset') or
+                            casematch(type, b'stream') or
+                            casematch(type, b'hash'))
+                if not is_valid:
+                    raise SimpleError(SYNTAX_ERROR_MSG)
```

Comment:
Delete this check. The real redis server doesn't check it.

Reply:
Removed.